homeassistant/components/habitica/notify.py

```diff
@@ -0,0 +1,208 @@
+    PRIVATE_MESSAGE = "private_message"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HabiticaConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the notify entity platform."""
+    members_added: set[UUID] = set()
+    entity_registry = er.async_get(hass)
+
+    coordinator = config_entry.runtime_data
+
+    if party := coordinator.data.user.party.id:
+        party_coordinator = hass.data[HABITICA_KEY][party]
+        async_add_entities(
+            [HabiticaPartyChatNotifyEntity(coordinator, party_coordinator.data.party)]
+        )
+
+        @callback
+        def add_entities() -> None:
+            nonlocal members_added
+
+            new_members = set(party_coordinator.data.members.keys()) - members_added
+            if TYPE_CHECKING:
+                assert coordinator.data.user.id
+            new_members.discard(coordinator.data.user.id)
+            if new_members:
+                async_add_entities(
+                    [
+                        HabiticaPrivateMessageNotifyEntity(
+                            coordinator, party_coordinator.data.members[member]
+                        )
+                        for member in new_members
+                    ]
+                )
+                members_added |= new_members
+
+            delete_members = members_added - set(party_coordinator.data.members.keys())
+            for member in delete_members:
+                if entity_id := entity_registry.async_get_entity_id(
+                    NOTIFY_DOMAIN,
+                    DOMAIN,
+                    f"{coordinator.config_entry.unique_id}_{member!s}_{HabiticaNotify.PRIVATE_MESSAGE}",
+                ):
+                    entity_registry.async_remove(entity_id)
+
+                members_added.discard(member)
+
+        party_coordinator.async_add_listener(add_entities)
+        add_entities()
+
+
+class HabiticaBaseNotifyEntity(HabiticaBase, NotifyEntity):
+    """Habitica base notify entity."""
+
+    entity_description: NotifyEntityDescription
+
+    def __init__(
+        self,
+        coordinator: HabiticaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize a Habitica entity."""
+        super().__init__(coordinator, self.entity_description)
+
+    @abstractmethod
+    async def _send_message(self, message: str) -> None:
+        """Send a Habitica message."""
+
+    async def async_send_message(self, message: str, title: str | None = None) -> None:
+        """Send a message."""
+        if TYPE_CHECKING:
+            assert self.entity_description.translation_placeholders
+        try:
+            await self._send_message(message)
+        except NotAuthorizedError as e:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="send_message_forbidden",
+                translation_placeholders={
+                    **self.entity_description.translation_placeholders,
```

Comment:
I can't seem to find the entity description here, I should probably go to bed. In any case, this kinda feels like misusing the translation placeholders, which I wouldn't be a fan of tho